Admonitions in Sphinx 4.4.0 lack closing div

### Describe the bug

Admonitions in Sphinx 4.4.0 with MyST lack a closing div. As a result, all content after the admonition becomes part of the admonition content.

Downgrading Sphinx to 4.3.2 resolves the issue.

### How to Reproduce

````
```{note}
This is my note
```

This content gets pulled into the note.

````

### Expected behavior

_No response_

### Your project

https://github.com/plone/documentation

### Screenshots

_No response_

### OS

macOS 10.15

### Python version

3.10.1

### Sphinx version

4.4.0

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

_No response_

> > Also used docutils 0.18.1 for all scenarios.
> 
> sphinx is specifically pinned to `<0.18`, so that could be an issue:

Yup, that was it.

PyCharm allows me to bypass pip's dependency resolver and install conflicting versions of packages.

![Screen Shot 2022-01-19 at 5 16 52 PM](https://user-images.githubusercontent.com/102112/150244914-e1382410-6e62-4636-a26d-34ab0128a5c2.png)

![Screen Shot 2022-01-19 at 5 17 09 PM](https://user-images.githubusercontent.com/102112/150244856-c3f607ab-2ad6-431f-95d2-7b6fb283d172.png)

But when I install the packages without such foolery, for example `pip install -r requirements.txt`, then the resolver prevents said foolery.

```
ERROR: Cannot install docutils==0.18.1 and sphinx==4.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested docutils==0.18.1
    sphinx 4.4.0 depends on docutils<0.18 and >=0.14

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
make: *** [build] Error 1
```

Wow. That is a serious footgun in PyCharm. Reported in:

https://youtrack.jetbrains.com/issue/PY-52623

In any case, this might be something to look out for in the future when you start using a later version of docutils. Here's a screenshot to illustrate the issue.

![Screen Shot 2022-01-19 at 5 20 34 PM](https://user-images.githubusercontent.com/102112/150244968-23e0ec54-04e8-4ac8-b2fe-854e1aafae1b.png)

And here's a repo where I was able to demonstrate and troubleshoot.

https://github.com/stevepiercy/sphinx-myst-admonition

See the README.md for installation instructions for how to replicate.

Feel free to close this issue or add it to the future todos.

--- a/sphinx/writers/html.py
+++ b/sphinx/writers/html.py
@@ -14,7 +14,7 @@ import posixpath
 import re
 import urllib.parse
 import warnings
-from typing import TYPE_CHECKING, Iterable, Tuple, cast
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, cast
 
 from docutils import nodes
 from docutils.nodes import Element, Node, Text
@@ -284,6 +284,9 @@ class HTMLTranslator(SphinxTranslator, BaseTranslator):
         if name:
             node.insert(0, nodes.title(name, admonitionlabels[name]))
         self.set_first_last(node)
+
+    def depart_admonition(self, node: Optional[Element] = None) -> None:
+        self.body.append('</div>\n')
 
     def visit_seealso(self, node: Element) -> None:
         self.visit_admonition(node, 'seealso')

--- a/sphinx/writers/html5.py
+++ b/sphinx/writers/html5.py
@@ -13,7 +13,7 @@ import posixpath
 import re
 import urllib.parse
 import warnings
-from typing import TYPE_CHECKING, Iterable, Set, Tuple, cast
+from typing import TYPE_CHECKING, Iterable, Optional, Set, Tuple, cast
 
 from docutils import nodes
 from docutils.nodes import Element, Node, Text
@@ -258,6 +258,9 @@ class HTML5Translator(SphinxTranslator, BaseTranslator):
             node, 'div', CLASS=('admonition ' + name)))
         if name:
             node.insert(0, nodes.title(name, admonitionlabels[name]))
+
+    def depart_admonition(self, node: Optional[Element] = None) -> None:
+        self.body.append('</div>\n')
 
     def visit_seealso(self, node: Element) -> None:
         self.visit_admonition(node, 'seealso')

